### PR TITLE
feat: Image Slider

### PR DESCRIPTION
**Pull request by dlnr**
The image slider is a work in progress since the Figma is nog yet finished. This PR focusses on  functionality and some groundwork. 

## What

This image slider uses `ImageSlider.Item` components to create a gallery that scrolls horizontally.

## How

I've used an intersectionObserver and native scroll to achieve a smooth gallery. The thumbnails are optional as well as scrollbar (you might want to hide them for presentational sites because the are ugly) and snapstop to prevent passing over the image.

## Checklist

- [x] Add or update unit tests
- [ ] Add or update documentation
- [x] Add or update stories
- [x] Add or update exports in index.\* files
- [x] Start the PR title with a Conventional Commit prefix, [as explained here](https://github.com/Amsterdam/design-system/blob/main/documentation/publishing.md?plain=1#L11).

## Additional notes

- A11y might need work, the thumbnails might want to have a label like "Item 2: <image-alt>". Extracting the image attributes felt kind of dodgy (cloneElement or change type to HTMLElement)
- Thumbnails scale between 58 an 188 pixels, that might change when the design is clear
- Thumnails scroll when there are too many and the don't fit withing the width provided by the grid.
- Thumbnails have no hover or focus event
- Thumnails are now button that contain a clone of the slides. This was a quick win, you might want to create a dedicated thumbnails with labels en a background image.

### CODE REVIEW

**Comment by RubenSibon** [comment]
Is an option to loopback to the start when you've reached the end desirable, i.e. "infinite rotation"? Or are there a11y issues with that?

What about an optional autoplay feature that can be turned on or off by the user?

**Comment by dlnr** [comment]
> Is an option to loopback to the start when you've reached the end desirable, i.e. "infinite rotation"? Or are there a11y issues with that?
> 
> What about an optional autoplay feature that can be turned on or off by the user?

The first is possible and it would eliminate the need to disable the button. It would visibly (smooth) scroll back to the start, which would be a little too much animation. I've used native horizontal scroll.

The autoplay feature would be a script and also possible. There are no specs for this though, the only feedback from design is that the thumbnails need to always fill the entire width of the gallery and are limited to 3 to 5 items.

**Review by VincentSmedinga — COMMENTED** [review]
Not sure my review is complete, but let me publish these comments that have been drafts for a couple of days.

File: packages/css/src/components/image-slider/image-slider.scss [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  gap: var(--ams-image-slider-gap);
+  grid-template-rows: 1fr auto;
+}
+
+.ams-image-slider__item {
+  scroll-snap-align: center;
+
+  /** temporary fix for covering the entire gallery */
+  .ams-image {
+    inline-size: 100%;
```
[the remark this answers is not in the document]
Comment (dlnr) [reply packages/css/src/components/image-slider/image-slider.scss:17]: This makes sure the image fills the image slider horizontally, also when the image resolution is lower than the width of the gallery.

File: packages/css/src/components/image-slider/image-slider.scss [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+.ams-image-slider__control {
+  place-self: center;
+  z-index: 1;
+}
+
+.ams-image-slider__thumbnails {
+  display: grid;
+  gap: var(--ams-image-slider-thumbnails-gap);
+  grid-auto-columns: minmax(58px, 188px);
```
[the remark this answers is not in the document]
Comment (dlnr) [reply packages/css/src/components/image-slider/image-slider.scss:66]: Removed in new PR

**Pull request closed without merging**